src/video/SDL_sysvideo.h

```diff
@@ -53,9 +53,17 @@ struct SDL_Window
     SDL_DisplayID last_fullscreen_exclusive_display;  /* The last fullscreen_exclusive display */
     SDL_DisplayID last_displayID;
 
-    /* Stored position and size for windowed mode */
+    /* Stored position and size for non-fullscreen windowed mode */
     SDL_Rect windowed;
 
+    /* Stored position and size for the window when not in a fixed-size state */
+    SDL_Rect floating;
```

Comment:
I'm confused, what's the difference between `windowed` and `floating` here? Which one should I use in video backend code?

Reply:
`windowed` is the size and position whenever the window is not in the fullscreen state. `floating` is base size of the window when it doesn't have some special system imposed size/restrictions on it such as maximized or tiled. If you are leaving fullscreen, the `windowed` size is used, if you are restoring the window from a state such as maximized or tiled, the floating size is used.

Basically `SDL_RestoreWindow()` should use the floating size/position, returning from fullscreen uses `windowed`. I'll better document this.

I wound up have having to track this in every backend, so it made sense to push it to the video layer. Having it in the video layer also lets us fix the long-standing case where a window would be created with the maximized flag, recreated by a renderer, and, since the last windowed size was used as the base size for recreating the window, unsetting the maximized state resulted in the window not returning to the expected size. Since the base floating size and the current windowed size are now separate, the recreated maximized window uses the correct original size and will be the expected size when restored.

Reply:
Much improved documentation, thanks!